policies/policies.go

```diff
@@ -148,4 +154,10 @@ type Repository interface {
 
 	// RetrieveAllDatasetsByOwner retrieves the subset of Datasets owned by the specified user
 	RetrieveAllDatasetsByOwner(ctx context.Context, ownerID string, pm PageMetadata) (PageDataset, error)
+
+	// InactivateDatasetByID inactivate a dataset
+	InactivateDatasetByID(ctx context.Context, sinkID string, ownerID string) error
+
+	// DeleteSinkFromDataset removes a sink from a dataset
+	DeleteSinkFromDataset(ctx context.Context, sinkID string, ownerID string) ([]Dataset, error)
```

Comment:
This actually removes the sinkID from _all_ Datasets, right? So the function should be called `DeleteSinkFromAllDatasets`